From 7e9ba5312a941ec13ec900989c4a8e229ff034c9 Mon Sep 17 00:00:00 2001
From: Sergey Edunov <edunov@fb.com>
Date: Wed, 20 Jul 2016 10:20:36 -0700
Subject: [PATCH] GIRAPH-1098 Job may get stuck if zookeeper port fixed and is
 in use

Test Plan: mvn clean verify -Phadoop_facebook

Reviewers: majakabiljo, dionysis.logothetis, maja.kabiljo

Reviewed By: maja.kabiljo

Differential Revision: https://reviews.facebook.net/D60945
---
 .../org/apache/giraph/zk/InProcessZooKeeperRunner.java   | 9 ++-------
 .../main/java/org/apache/giraph/zk/ZooKeeperManager.java | 2 +-
 .../main/java/org/apache/giraph/zk/ZooKeeperRunner.java  | 4 +++-
 3 files changed, 6 insertions(+), 9 deletions(-)

diff --git a/giraph-core/src/main/java/org/apache/giraph/zk/InProcessZooKeeperRunner.java b/giraph-core/src/main/java/org/apache/giraph/zk/InProcessZooKeeperRunner.java
index 4f15f3a22..cee2c7885 100644
--- a/giraph-core/src/main/java/org/apache/giraph/zk/InProcessZooKeeperRunner.java
+++ b/giraph-core/src/main/java/org/apache/giraph/zk/InProcessZooKeeperRunner.java
@@ -47,13 +47,8 @@ public class InProcessZooKeeperRunner
   private QuorumRunner quorumRunner = new QuorumRunner();
 
   @Override
-  public int start(String zkDir, final ZookeeperConfig config) {
-    try {
-      return quorumRunner.start(config);
-    } catch (IOException e) {
-      LOG.error("Unable to start zookeeper", e);
-    }
-    return -1;
+  public int start(String zkDir, ZookeeperConfig config) throws IOException {
+    return quorumRunner.start(config);
   }
 
   @Override
diff --git a/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperManager.java b/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperManager.java
index 097172d27..b8438821b 100644
--- a/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperManager.java
+++ b/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperManager.java
@@ -457,7 +457,7 @@ private void generateZooKeeperConfig() {
    * If this task has been selected, online a ZooKeeper server.  Otherwise,
    * wait until this task knows that the ZooKeeper servers have been onlined.
    */
-  public void onlineZooKeeperServer() {
+  public void onlineZooKeeperServer() throws IOException {
     if (zkServerTask == taskPartition) {
       File zkDirFile = new File(this.zkDir);
       try {
diff --git a/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperRunner.java b/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperRunner.java
index 279704713..9d6a33220 100644
--- a/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperRunner.java
+++ b/giraph-core/src/main/java/org/apache/giraph/zk/ZooKeeperRunner.java
@@ -19,6 +19,8 @@
 
 import org.apache.giraph.conf.ImmutableClassesGiraphConfigurable;
 
+import java.io.IOException;
+
 /**
  * ZooKeeper wrapper interface.
  * Implementation should provide a way to start, stop and cleanup
@@ -33,7 +35,7 @@ public interface ZooKeeperRunner extends ImmutableClassesGiraphConfigurable {
    * @param config zookeeper configuration
    * @return port zookeeper runs on
    */
-  int start(String zkDir, ZookeeperConfig config);
+  int start(String zkDir, ZookeeperConfig config) throws IOException;
 
   /**
    * Stops zookeeper.
